contracts/protocols/arbitrum/aave_v3/helpers.sol

```diff
@@ -190,55 +199,62 @@ contract AaveV3Helper is DSMath {
     IAaveProtocolDataProvider internal aaveData = IAaveProtocolDataProvider(provider.getPoolDataProvider());
     IPool internal pool = IPool(provider.getPool());
     IUiIncentiveDataProviderV3 internal uiIncentives = IUiIncentiveDataProviderV3(getUiIncetivesProvider());
+    IRewardsController internal rewardsCntr = IRewardsController(getRewardsController());
+
+    function getUserReward(
+        address user,
+        address[] memory assets,
+        RewardsInfo[] memory _rewards
+    ) internal view returns (UserRewards memory unclaimedRewards) {
+        if (_rewards.length > 0) {
+            (address[] memory reserves, uint256[] memory rewards) = rewardsCntr.getAllUserRewards(assets, user);
+            unclaimedRewards = UserRewards(reserves, rewards);
+        }
+    }
 
     function getIncentivesInfo(address user) internal view returns (ReserveIncentiveData[] memory incentives) {
         AggregatedReserveIncentiveData[] memory _aggregateIncentive = uiIncentives.getReservesIncentivesData(provider);
-        UserReserveIncentiveData[] memory _aggregateUserIncentive = uiIncentives.getUserReservesIncentivesData(
-            provider,
-            user
-        );
         incentives = new ReserveIncentiveData[](_aggregateIncentive.length);
         for (uint256 i = 0; i < _aggregateIncentive.length; i++) {
+            address[] memory rToken = new address[](1);
             RewardsInfo[] memory _aRewards = getRewardInfo(
-                _aggregateIncentive[i].aIncentiveData.rewardsTokenInformation,
-                _aggregateUserIncentive[i].aTokenIncentivesUserData.userRewardsInformation
+                _aggregateIncentive[i].aIncentiveData.rewardsTokenInformation
             );
             RewardsInfo[] memory _sRewards = getRewardInfo(
-                _aggregateIncentive[i].sIncentiveData.rewardsTokenInformation,
-                _aggregateUserIncentive[i].sTokenIncentivesUserData.userRewardsInformation
+                _aggregateIncentive[i].sIncentiveData.rewardsTokenInformation
             );
             RewardsInfo[] memory _vRewards = getRewardInfo(
-                _aggregateIncentive[i].vIncentiveData.rewardsTokenInformation,
-                _aggregateUserIncentive[i].vTokenIncentivesUserData.userRewardsInformation
+                _aggregateIncentive[i].vIncentiveData.rewardsTokenInformation
             );
+            rToken[0] = _aggregateIncentive[i].aIncentiveData.tokenAddress;
```

Comment:
Why don't you pass this directly?

Reply:
The function from the RewardsController, `getAllUserRewards` takes array of assets as input.

Reply:
Ohh okay